src/main/groovy/com/twcable/grabbit/jcr/AuthorizableProtoNodeDecorator.groovy

```diff
@@ -0,0 +1,277 @@
+        }
+        catch(SecurityException ex) {
+            issues << 'accessDeclaredMembers'
+            badPermissions = true
+        }
+        try {
+            //Needed to access classes directly within a potentially system protected package
+            securityManager.checkPermission(new RuntimePermission('accessClassInPackage.{org.apache.jackrabbit.oak.security.user}'))
+        }
+        catch(SecurityException ex) {
+            issues << 'accessClassInPackage.{org.apache.jackrabbit.oak.security.user}'
+            badPermissions = true
+        }
+        if(badPermissions) {
+            log.warn "A SecurityManager is enabled for this JVM, and permissions are not sufficient for Grabbit to sync Authorizables (Users/Groups). You must enable ${issues.join(', ')} permissions in your SecurityManager to use this functionality" +
+                     "Check https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimePermission.html and https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/ReflectPermission.html to see what these permissions enable"
+            return false
+        }
+        else {
+            log.debug "Permissions check successful"
+            return true
+        }
+    }
+
+    /**
+     * Mostly for ease of mocking/testing
+     * @return the system's security manager, or null if one is not present
+     */
+    SecurityManager getSecurityManager() {
+        return System.getSecurityManager()
+    }
+
+    /**
+     * Mostly for ease of mocking/testing
+     */
+    UserManager getUserManager(final Session session) {
+        return AccessControlUtil.getUserManager(session)
+    }
+
+
+    /**
+     * Normally we would call org.apache.jackrabbit.oak.jcr.delegate.UserDelegator.changePassword(String password) to change a password (this is what is publicly available through the Jackrabbit API)
+     * However, this method ALWAYS rehashes the password argument which is of no use to us, since we are trying to transfer an already hashed password.
+     *
+     * Internally, org.apache.jackrabbit.oak.jcr.delegate.UserDelegator calls it's delegate's org.apache.jackrabbit.oak.security.user.UserImpl.changePassword(String password)
+     * which calls org.apache.jackrabbit.oak.security.user.UserManagerImpl.setPassword(Tree tree, String userId, String password, boolean forceHash) with forceHash always set to true
+     * We really need forcehash set to false for our case, but this isn't publicly available. Here, we access internal objects to do this manipulation. org.apache.jackrabbit.oak.security.user.UserManagerImpl
+     * simply ensures that forcehash is false, and that the password is not plain text, and it sets the password as-is.
+     *
+     * @throws IllegalStateException if security permissions required to run this are not there. @{code checkSecurityPermissions()} should be called before calling this method
+     **/
+     void setPasswordForUser(final User user, final Session session) {
```

Comment:
Oh, and by the way. To remove this in the future, we have : 

https://github.com/apache/jackrabbit/pull/45